File: instance_gen.cpp
#include <iostream>
#include <time.h>
#include <cstdlib>
#include <fstream>
#define LO 0.2
#define HI 0.5
#define MIN_TASK 50
#define MAX_TASK 10000
#define MIN_TIME 20
#define MAX_TIME 500
using namespace std;

int main(){
    srand(time(NULL));
    // m - liczba procesorow
    // n - liczba zadan
    int m, n;
    // wspolczynnik fract = m/n
    float fract = LO + (float) (rand()) / ((float) (RAND_MAX / (HI - LO))); // (LO, Hi)

    // liczba zadan 50 < n < 10000
    n = rand() % (MAX_TASK - MIN_TASK) + MIN_TASK;
    m = (int) (fract * n);

    fstream f;
    f.open("file1", ios::out);
    if (!f) { cerr << "File not opened" << endl; }

    f << m << endl << n << endl;

    // czasy zadan 20 << n << 500
    for (int i = 0; i < n; i++) {
        m = rand() % (MAX_TIME - MIN_TIME) + MIN_TIME;
        f << m << endl;
    }

    f.close();

    return 0;
}

File: instance_gen.py
import os
import random
import subprocess

LO = 0.2
HI = 0.5
MIN_TASK = 50
MAX_TASK = 10000
MIN_TIME = 20
MAX_TIME = 500
N_INSTANCES = 10
INST_DIR = 'instances'


def mkdir(dir):
    try:
        os.mkdir(dir)
    except:
        pass

mkdir(INST_DIR)

for i in range(1, N_INSTANCES+1):
    fract = random.uniform(LO, HI)
    task_n = random.randint(MIN_TASK, MAX_TASK)
    proc_n = int(fract * task_n)

    f = open('{}/instance_{}'.format(INST_DIR, i), 'w')
    f.write('{}\n{}\n'.format(proc_n, task_n))

    for i in range(task_n):
        proc_n = random.randint(MIN_TASK, MAX_TASK)
        f.write('{}\n'.format(proc_n))
f.close()
